CONTRIBUTING.md

```diff
@@ -39,6 +39,38 @@ request into the body of the commit message.
 
 If the Travis tests are failing on your branch, you should look at the logs to figure out why. Sometimes they fail spuriously, in which case you can post a comment requesting that a project owner kick the build.
 
+# Error handling
+
+All errors must be addressed in some way: That may be simply by returning an
+error up the stack, or by handling it in some intelligent way where it is
+generated, or explicitly ignored by assigning it to `_`. We use the `errcheck`
+tool in our integration tests to make sure all errors are addressed. Note that
+even in tests, errors should not be ignored, since they may generate
+hard-to-debug problems.
+
+We define two special types of error. BoulderErrors, defined in
+errors/errors.go, are used specifically when an typed error needs to be passed
+across an RPC boundary. For instance, if the SA returns "not found", callers
+need to be able to distinguish that from a network error. Not every error that
+may pass across an RPC boundary needs to be a BoulderError, only those errors
+that need to be handled by type elsewhere. Handling by type may be as simple as
+turning a BoulderError into a specific type of ProblemDetail.
+
+The other special type of error is ProblemDetail. We try to treat these as a
+presentation-layer detail, and use them only in parts of the system that are
+responsible for rendering ProblemDetails to end-users, i.e. wfe and wfe2. Note
```

Comment:
I think it's confusing to explain ProblemDetails as being the domain of the thing that renders ProblemDetails. Maybe -> "and use them only in parts of the system that are responsible for rendering error information to API clients/end users"